src/common/transformations/src/transformations/common_optimizations/lstm_cell_fusion.cpp

```diff
@@ -23,6 +23,107 @@
+    auto x_by_wi_biased = pattern::wrap_type<op::v1::Add>({x_by_wi, bi});
+    h_by_ri = pattern::wrap_type<op::v0::MatMul>({h, ri});
+    it = pattern::wrap_type<op::v1::Add>({x_by_wi_biased, h_by_ri});
+    it = pattern::wrap_type<op::v0::Relu, op::v0::Sigmoid, op::v0::Tanh>({it});
+}
+
+bool check_weights_format(const ov::Output<ov::Node>& w,
+                          const ov::Output<ov::Node>& r,
+                          const ov::Output<ov::Node>& b,
+                          size_t input_size,
+                          size_t hidden_size,
+                          const ov::pass::pattern::PatternValueMap& pattern_map,
+                          const std::shared_ptr<Node>& x_by_w_label,
+                          const std::shared_ptr<Node>& h_by_r_label) {
+    // w must be of a shape [input_size, hidden_size]
+    // r must be of a shape [hidden_size, hidden_size]
+    // b must be of a shape [hidden_size]
+    if (w.get_shape() != ov::Shape{input_size, hidden_size}) {
+        return false;
+    }
+
+    if (r.get_shape() != ov::Shape{hidden_size, hidden_size}) {
+        return false;
+    }
+
+    if (b.get_shape() != ov::Shape{hidden_size}) {
+        return false;
+    }
+
+    // check transpose attributes for MatMul operations
+    if (const auto& matmul = ov::as_type_ptr<op::v0::MatMul>(pattern_map.at(x_by_w_label).get_node_shared_ptr())) {
+        if (matmul->get_transpose_a() || matmul->get_transpose_b()) {
+            return false;
+        }
+    } else {
+        return false;
+    }
+
+    if (const auto& matmul = ov::as_type_ptr<op::v0::MatMul>(pattern_map.at(h_by_r_label).get_node_shared_ptr())) {
+        if (matmul->get_transpose_a() || matmul->get_transpose_b()) {
+            return false;
+        }
+    } else {
+        return false;
+    }
+
+    return true;
+}
+
+ov::Output<ov::Node> prepare_weight_fico(const ov::Output<ov::Node>& f,
+                                         const ov::Output<ov::Node>& i,
+                                         const ov::Output<ov::Node>& c,
+                                         const ov::Output<ov::Node>& o,
+                                         NodeRegistry& rg) {
+    // at this point input weights are of shape [input_size, hidden_size]
+    // before concatenation it needs to transpose them
+    // to get a shape equal to [hidden_size, input_size]
+    auto transpose_order = rg.make<ov::op::v0::Constant>(element::i32, ov::Shape{2}, std::vector<int32_t>{1, 0});
+    auto f_tr = rg.make<ov::op::v1::Transpose>(f, transpose_order);
+    auto i_tr = rg.make<ov::op::v1::Transpose>(i, transpose_order);
+    auto c_tr = rg.make<ov::op::v1::Transpose>(c, transpose_order);
+    auto o_tr = rg.make<ov::op::v1::Transpose>(o, transpose_order);
+
+    ov::Output<ov::Node> w = rg.make<ov::op::v0::Concat>(ov::OutputVector{f_tr, i_tr, c_tr, o_tr}, 0);
+    if (const auto& constant = ov::util::constantfold_subgraph(w)) {
+        w = constant;
```

Comment:
There is a ConstantFolding transformation in MOC. Do we need it here?

Reply:
Yes